seqr/views/utils/airflow_utils.py

```diff
@@ -30,75 +33,112 @@ def trigger_airflow_data_loading(*args, user: User, individual_ids: list[int], s
     upload_info = [f'Pedigree files have been uploaded to {gs_path}']
 
     try:
-        _check_dag_running_state()
-        _update_variables(updated_variables)
-        _wait_for_dag_variable_update(updated_variables['projects_to_run'])
-        _trigger_dag()
+        _check_dag_running_state(LOADING_PIPELINE_DAG_NAME)
+        _update_variables(updated_variables, LOADING_PIPELINE_DAG_NAME)
+        _wait_for_dag_variable_update_via_tasks(updated_variables['projects_to_run'], LOADING_PIPELINE_DAG_NAME)
+        _trigger_dag(LOADING_PIPELINE_DAG_NAME)
     except Exception as e:
         logger_call = logger.warning if isinstance(e, DagRunningException) else logger.error
         logger_call(str(e), user)
-        _send_slack_msg_on_failure_trigger(e, updated_variables, error_message)
+        _send_slack_msg_on_failure_trigger(LOADING_PIPELINE_DAG_NAME, e, updated_variables, error_message)
         success = False
 
     if success or success_slack_channel != SEQR_SLACK_LOADING_NOTIFICATION_CHANNEL:
-        _send_load_data_slack_msg([success_message] + upload_info, success_slack_channel, updated_variables)
+        _send_load_data_slack_msg(
+            LOADING_PIPELINE_DAG_NAME, [success_message] + upload_info, success_slack_channel, updated_variables
+        )
     return success
 
 
-def _send_load_data_slack_msg(messages: list[str], channel: str, dag: dict):
+def trigger_airflow_delete_families(
+    dataset_type: str, family_guids: list[str], from_project: Project,
+):
+    success = True
+    variables = {
+        'projects_to_run': [from_project.guid],
+        'family_guids': sorted(family_guids),
+        'reference_genome': reference_genome_version(from_project.genome_version),
+        'dataset_type': dag_dataset_type(dataset_type),
+    }
+    try:
+        _check_dag_running_state(DELETE_FAMILIES_DAG_NAME)
+        _update_variables(variables, DELETE_FAMILIES_DAG_NAME)
+        _wait_for_dag_variable_update(variables, DELETE_FAMILIES_DAG_NAME)
+        _trigger_dag(DELETE_FAMILIES_DAG_NAME)
+    except Exception as e:
+        logger_call = logger.warning if isinstance(e, DagRunningException) else logger.error
+        logger_call(str(e))
+        success = False
+
+    if success:
+        logger.info(f'Successfully triggered DELETE_FAMILIES DAG for {len(family_guids)} family from {from_project.name}/{dataset_type}')
```

Comment:
rather than setting a success variable, just move this to the bottom of the try block - if an exception does happen it won't be called